kgfx.cpp

```diff
@@ -187,49 +184,249 @@ void KGFX::drawChartLarge(std::vector<float> arr, int color, int y, int height)
+  if (val > min) {
+    tft.drawSmoothArc(center, center, innerR, 80, 70 + angle, 290, bg_color, bg_color, true);
+    tft.drawSmoothArc(center, center, innerR, 80, 70, 70 + angle, fg_color, bg_color, true);
+    if (haskMarks) {
+      draw_round_hash_mark((center - width) - markLen, center - width, (center - width) - shortMarkLen, center - width);
+    }
+  } else { // draw an empty arc
+    tft.drawSmoothArc(center, center, innerR, 80, 70, 290, fg_color, fg_color, true);
+    if (hashMarks) {
+      draw_round_hash_mark((center - width) - markLen, center - width, (center - width) - shortMarkLen, center - width);
+    }
+  }
+  String f = String(val).c_str();
+  char buffer[f.length() + 2 + strlen(suffix)];
+  sprintf(buffer, "%s%s", f, suffix);
+  drawTextCenter(buffer, Arial_24_Bold, fg_color, center - 10); // value
+  drawText(String(min).c_str(), Arial_12, bg_color, 25, 170);   // min value
+  drawText(String(max).c_str(), Arial_12, bg_color, 180, 170);  // max value
+  drawTextCenter(label, Arial_24_Bold, fg_color, 200);          // label
+}
+
+/***************************************************************************************
+ * Function name:           drawGauge
+ * Description:             Draws gauge with unsigned long value
+ * @param label            Label for the gauge
+ * @param suffix           Suffix for the gauge stats
+ * @param val              Value for the gauge
+ * @param max              Maximum value for the gauge
+ * @param min              Minimum value for the gauge
+ * @param bg_color         Background color for the gauge
+ * @param fg_color         Foreground color for the gauge
+ * @param hashMarks        Draw hash marks on the gauge
+ * ***************************************************************************************/
+void KGFX::drawGauge(const char *label, const char *suffix, unsigned long val, int max, int min, int bg_color, int fg_color, bool hashMarks) {
+  int center = 120;            // center
+  int width = 20;              // width
+  int innerR = center - width; // inner radius
+  double breakPoint = (double)(val - min) / (max - min);
+  double angle = 220 * breakPoint;
+  int shortMarkLen = ((center - width) * 5 / 6) / 6;
+  int markLen = ((center - width) * 5 / 6) / 2;
+  constexpr float deg2rad = 3.14159265359 / 220;
+  if (val > min) {
+    tft.drawSmoothArc(center, center, innerR, 80, 70 + angle, 290, bg_color, bg_color, true);
+    tft.drawSmoothArc(center, center, innerR, 80, 70, 70 + angle, fg_color, bg_color, true);
+    if (haskMarks) {
+      draw_round_hash_mark((center - width) - markLen, center - width, (center - width) - shortMarkLen, center - width);
+    }
+  } else { // draw an empty arc
+    tft.drawSmoothArc(center, center, innerR, 80, 70, 290, fg_color, fg_color, true);
+    if (hashMarks) {
+      draw_round_hash_mark((center - width) - markLen, center - width, (center - width) - shortMarkLen, center - width);
+    }
+  }
+  String f = String(val).c_str();
```

Comment:
Since the code above is the same across the new methods, can you write a function for the duplicate parts?

Reply:
To be honest, I thought the same thing but was then too lazy to do it. :-) I'll fix that.

Reply:
created a function for these.